tock/employees/models.py

```diff
@@ -131,6 +134,29 @@ def organization_name(self):
 
         return ''
 
+    def is_late(self):
+        """
+        Checks if user has a timecard submitted for the
+        most recent Reporting Period.
+
+        We're using get_model() to avoid circular imports
+        since so many things use UserData
+        """
+        RP_model = apps.get_model('hours', 'ReportingPeriod')
+        TC_model = apps.get_model('hours', 'Timecard')
+        rp = RP_model.objects.order_by('end_date').filter(
+            end_date__lt=datetime.date.today()
+        ).latest()
+        timecard_count = TC_model.objects.filter(
+                reporting_period=rp,
+                submitted=True,
+                user=self.user,
+            ).count()
+        if timecard_count is 0:
+            return True
+        else:
+            return False
```

Comment:
This looks good - filter out all timecards that match the user, reporting period and if they've been submitted; if none are returned, we know they're late. 👍

Reply:
We don't have this logic elsewhere in the code, do we?  It seems familiar, perhaps in one of the views. I wonder if this could serve to replace it, or if perhaps it should be abstracted out as a utility method that both this model method and the logic in the view call?

Reply:
We'll revisit this as a later time, there's no need to hold things up here with it!

Reply:
We may also want to extract this in the future to provide support for crafting manager methods that allow us to filter query sets specifically around whether or not someone is late.